### ISSUE TITLE
meson build fails

### ISSUE THREAD

**Issue by Tetsujinfr** [meghprkh/libgamepad#1]
hi, not sure what I did wrong. I installed all the dependencies, then I am doing the following (I am on a Jetson with Jetpack 4.4, with ARM64 CPUs):

<pre><font color="#8AE234"><b>tetsfr@tets</b></font>:<font color="#729FCF"><b>/media/tetsfr/SSD</b></font>$ git clone --recurse-submodules  https://github.com/meghprkh/libgamepad.git
Cloning into &apos;libgamepad&apos;...
remote: Enumerating objects: 405, done.
remote: Counting objects: 100% (405/405), done.
remote: Compressing objects: 100% (157/157), done.
remote: Total 405 (delta 229), reused 405 (delta 229), pack-reused 0
Receiving objects: 100% (405/405), 84.52 KiB | 739.00 KiB/s, done.
Resolving deltas: 100% (229/229), done.
<font color="#8AE234"><b>tetsfr@tets</b></font>:<font color="#729FCF"><b>/media/tetsfr/SSD</b></font>$ cd libgamepad/
<font color="#8AE234"><b>tetsfr@tets</b></font>:<font color="#729FCF"><b>/media/tetsfr/SSD/libgamepad</b></font>$ mkdir build
<font color="#8AE234"><b>tetsfr@tets</b></font>:<font color="#729FCF"><b>/media/tetsfr/SSD/libgamepad</b></font>$ meson build/

<b>The Meson build system</b>
Version: 0.54.3
Source dir: <b>/media/tetsfr/SSD/libgamepad</b>
Build dir: <b>/media/tetsfr/SSD/libgamepad/build</b>
Build type: <b>native build</b>
Project name: <b>LibGamepad</b>
Project version: <b>1.0.0</b>
C compiler for the host machine: <b>cc</b> (gcc 7.5.0 &quot;cc (Ubuntu/Linaro 7.5.0-3ubuntu1~18.04) 7.5.0&quot;)
C linker for the host machine: <b>cc</b> ld.bfd 2.30
Vala compiler for the host machine: <b>valac</b> (valac 0.40.19)
Host machine cpu family: <b>aarch64</b>
Host machine cpu: <b>aarch64</b>
Found pkg-config: <b>/usr/bin/pkg-config</b> (0.29.1)
Run-time dependency <b>glib-2.0</b> found: <font color="#8AE234"><b>YES</b></font> <font color="#06989A">2.56.4</font>
Run-time dependency <b>gobject-2.0</b> found: <font color="#8AE234"><b>YES</b></font> <font color="#06989A">2.56.4</font>
Run-time dependency <b>gio-2.0</b> found: <font color="#8AE234"><b>YES</b></font> <font color="#06989A">2.56.4</font>
Found pkg-config: <b>/usr/bin/pkg-config</b> (0.29.1)
Run-time dependency <b>gudev-1.0</b> found: <font color="#8AE234"><b>YES</b></font> <font color="#06989A">232</font>
Run-time dependency <b>libevdev</b> found: <font color="#8AE234"><b>YES</b></font> <font color="#06989A">1.5.8</font>

src/meson.build:64:0: <font color="#EF2929"><b>ERROR:</b></font> File /media/tetsfr/SSD/libgamepad/build/src/libgamepad.h does not exist.

A full log can be found at <b>/media/tetsfr/SSD/libgamepad/build/meson-logs/meson-log.txt</b>
</pre>

**Comment by meghprkh** [comment]
Hi, it seems that the vala compilation for meson has changed

https://mesonbuild.com/Vala.html#building-libraries

We need to build all three, a header, a GIR and a VAPI. I don't have access to a setup right now, but this shall be an easy change, let me know if you need help

More detailed steps afaik:
- Remove `vala_args` in [src/meson.build](src/meson.build)
- For additional VAPI search path of `/drivers/linux/vapis`, use `add_project_arguments` as mentioned [here](https://mesonbuild.com/Vala.html#adding-to-valas-search-path)
- Get rid of the manual `install`s and use `install_dir: [true, true, true, true])` as mentioned [here](https://mesonbuild.com/Vala.html#building-libraries)

**Comment by Tetsujinfr** [comment]
yes I would need more details on the steps to follow. I am not familiar with Meson, neither with Vala I am afraid. I do not know which path to add to the search with add_project_arguments and I do know how to use install_dir and where. I am quite confused on how to get it to compile properly.
thanks a lot for your help
